### ISSUE TITLE
System.Net.NetworkInformation.IPInterfaceProperties.DnsSuffix is not working as expected on Unix family OSes

### ISSUE THREAD

**Issue by onatm** [dotnet/corefx#41953]
Currently, `System.Net.NetworkInformation.IPInterfaceProperties.DnsSuffix` property returns only the `search` keyword's value when there is exactly one entry on the line where `search` is present.

According to [resolv.conf](https://linux.die.net/man/5/resolv.conf) manual, there could be multiple entries for a given search keyword.

Following the documentation, you can also see that the `domain` keyword could be used as the first search domain when specified.

Additionally, when both of the keywords are specified, the last one will be used as the search domain:

> The domain and search keywords are mutually exclusive. If more than one instance of these keywords is present, the last instance wins.

Returning multiple search domain entries (DNS suffixes) would break the backward compatibility. However, there should be a way for Unix family Oses to acquire this data somehow and the `domain` keyword should be used as well.

**Comment by onatm** [comment]
@davidsh could you also add `os-mac-os-x` label as well?

**Comment by wfurt** [comment]
I agree that `domain` should be supported and it probably should take precedence over `search` as well. 
As far as list of all domains DNS is willing to search: that would need new API. Also current structure assumes that this is per interface.  However that is not true in practice on Unix systems where DNS resolver is global per system (or container)

**Comment by wfurt** [comment]
note that OSX allows per interface configuration
```
networksetup -listallnetworkservices
networksetup -getsearchdomains <networkservice>
```
But we currently do not use that interface so we fall back to processing /etc/resolv.conf.
